clarity/enhancer/multiband_compressor/multiband_compressor.py

```diff
@@ -0,0 +1,259 @@
+            raise ValueError(
+                "Attack must be a float or have the same "
+                "length as crossover frequencies + 1. "
+                f"{len(attack)} was provided, {self.num_compressors} expected."
+            )
+        if len(release) != self.num_compressors:
+            raise ValueError(
+                "Release must be a float or have the same "
+                "length as crossover frequencies + 1. "
+                f"{len(release)} was provided, {self.num_compressors} expected."
+            )
+        if len(threshold) != self.num_compressors:
+            raise ValueError(
+                "Threshold must be a float or have the same "
+                "length as crossover frequencies + 1. "
+                f"{len(threshold)} was provided, {self.num_compressors} expected."
+            )
+        if len(ratio) != self.num_compressors:
+            raise ValueError(
+                "Ratio must be a float or have the same "
+                "length as crossover frequencies + 1. "
+                f"{len(ratio)} was provided, {self.num_compressors} expected."
+            )
+        if len(gain) != self.num_compressors:
+            raise ValueError(
+                "Gain must be a float or have the same length as "
+                "crossover frequencies + 1. "
+                f"{len(gain)} was provided, {self.num_compressors} expected."
+            )
+        if len(knee_width) != self.num_compressors:
+            raise ValueError(
+                "Knee width must be a float or have the same length as "
+                "crossover frequencies + 1. "
+                f"{len(knee_width)} was provided, {self.num_compressors} expected."
+            )
+
+        self.compressor = [
+            Compressor(
+                attack=attack[i],
+                release=release[i],
+                threshold=threshold[i],
+                ratio=ratio[i],
+                gain=gain[i],
+                knee_width=knee_width[i],
+                sample_rate=self.sample_rate,
+            )
+            for i in range(self.num_compressors)
+        ]
+
+    def __call__(
+        self, signal: np.ndarray, return_bands: bool = False
+    ) -> np.ndarray | tuple:
+        """Compress the signal.
+
+        Args:
+            signal (np.ndarray): The input signal.
+                (channel, samples)
+            return_bands (bool): If True, return the compressed bands.
+
+        Returns:
+            np.ndarray: The compressed signal.
+                (channel, samples)
+            np.ndarray: The compressed bands if return_bands is True.
+                (bands, channel, samples)
+        """
+
+        bands_signal = self.crossover(signal[0])[:, np.newaxis, :]
+        if signal.shape[0] > 1:
+            for sig in signal[1:]:
+                bands_signal = np.hstack(
+                    (bands_signal, self.crossover(sig)[:, np.newaxis, :])
+                )
+
```

Comment:
I don't think this loop should be necessary. Your crossover filter is based on lfilter and so should be able to apply the crossover to all the channels in a single call.   i.e. bands_signal = self.crossover(signal)[:, np.newaxis, :].  You might need to make sure that if the signal is single channel then an axis added. And you will need to modify crossover so that it filters along the time axes.